snapcraft/internal/repo/_deb.py

```diff
@@ -475,7 +475,12 @@ def fetch_stage_packages(
     def unpack_stage_packages(
         cls, *, stage_packages_path: pathlib.Path, install_path: pathlib.Path
     ) -> None:
-        for pkg_path in stage_packages_path.glob("*.deb"):
+        stage_packages = stage_packages_path.glob("*.deb")
+
+        if not stage_packages:
```

Comment:
This hunk do anything?

`if not stage_packages` will never be true AFAIK because glob() returns a generator

Reply:
Wasn't that iglob?

Reply:
yes. glob returns a list of paths, and iglob is returns an iterator.

Reply:
For completion
```
>>> import glob
>>> glob.glob("*.py")
['setup.py', 'units.py']
```

Reply:
There's a catch here, `glob.glob()` returns a list of paths but `Path.glob()` still returns a generator:
```
>>> import pathlib
>>> p = pathlib.Path("/")
>>> p.glob("*")
<generator object Path.glob at 0x7fa9a7356f20>
```

Reply:
@sergiusens may want to fix that in a follow-up.